notebook/static/base/js/utils.js

```diff
@@ -192,165 +193,231 @@ define([
         return uuid;
     };
 
-
-    //Fix raw text to parse correctly in crazy XML
-    function xmlencode(string) {
-        return string.replace(/\&/g,'&'+'amp;')
-            .replace(/</g,'&'+'lt;')
-            .replace(/>/g,'&'+'gt;')
-            .replace(/\'/g,'&'+'apos;')
-            .replace(/\"/g,'&'+'quot;')
-            .replace(/`/g,'&'+'#96;');
-    }
-
-
-    //Map from terminal commands to CSS classes
-    var ansi_colormap = {
-        "01":"ansibold",
-        
-        "30":"ansiblack",
-        "31":"ansired",
-        "32":"ansigreen",
-        "33":"ansiyellow",
-        "34":"ansiblue",
-        "35":"ansipurple",
-        "36":"ansicyan",
-        "37":"ansigray",
-        
-        "40":"ansibgblack",
-        "41":"ansibgred",
-        "42":"ansibggreen",
-        "43":"ansibgyellow",
-        "44":"ansibgblue",
-        "45":"ansibgpurple",
-        "46":"ansibgcyan",
-        "47":"ansibggray"
-    };
+    var _ANSI_COLORS = [
+        "ansi-black",
+        "ansi-red",
+        "ansi-green",
+        "ansi-yellow",
+        "ansi-blue",
+        "ansi-magenta",
+        "ansi-cyan",
+        "ansi-white",
+        "ansi-black-intense",
+        "ansi-red-intense",
+        "ansi-green-intense",
+        "ansi-yellow-intense",
+        "ansi-blue-intense",
+        "ansi-magenta-intense",
+        "ansi-cyan-intense",
+        "ansi-white-intense",
+    ];
     
-    function _process_numbers(attrs, numbers) {
-        // process ansi escapes
-        var n = numbers.shift();
-        if (ansi_colormap[n]) {
-            if ( ! attrs["class"] ) {
-                attrs["class"] = ansi_colormap[n];
+    function _parseNumbers(text) {
+        var numbers = [];
+        var items = text.split(";");
+        for (var i = 0; i < items.length; i++) {
+            var item = items[i];
+            if (item === "") {
+                numbers.push(0);
+            } else if (item.search(/^\d+$/) !== -1) {
+                numbers.push(parseInt(item));
             } else {
-                attrs["class"] += " " + ansi_colormap[n];
-            }
-        } else if (n == "38" || n == "48") {
-            // VT100 256 color or 24 bit RGB
-            if (numbers.length < 2) {
-                console.log("Not enough fields for VT100 color", numbers);
+                // Ignored: Invalid color specification
+                numbers.length = 0;
```

Comment:
I actually have to clear `numbers` here, because there might already be some entries from previous loop iterations.

Would you prefer if I used

```js
numbers = [];
```

Or something entirely different?